### ISSUE TITLE
you're accidentally DDoSing api.npmjs.org every hour 

### ISSUE THREAD

**Issue by bcoe** [emberobserver/client#83]
first off, https://emberobserver.com is really cool 😄 it's awesome to see people building such slick things on top of the npm registry.

_unfortunately_, your website was brought to my attention because your ip seems to be accidentally DDoSing api.npmjs.org every hour or so.

<img width="1035" alt="screen shot 2017-05-18 at 10 43 45 pm" src="https://cloud.githubusercontent.com/assets/194609/26234727/a6278012-3c1b-11e7-9330-9deb1ceca644.png">

I believe those spikes are the `npm-registry` backing emberobserver.com re-indexing download-counts for each package.

This is resulting in your site getting a `429` response for a huge # of requests each hour:

```
<134>2017-05-19T05:21:03Z cache-jfk8128 fastly-logs-5-east[285597]: 45.33.64.72 "-" "GET /downloads/point/last-day/ember-cli-deploy-build" 429 "npm-registry/0.1.13 node/v4.2.6" "" "MISS" "(null)" "cache-jfk8128-JFK" "381" "213"
<134>2017-05-19T05:21:03Z cache-jfk8128 fastly-logs-5-east[285597]: 45.33.64.72 "-" "GET /downloads/point/last-day/ember-cli-deploy-zookeeper" 429 "npm-registry/0.1.13 node/v4.2.6" "" "MISS" "(null)" "cache-jfk8128-JFK" "331" "213"
<134>2017-05-19T05:21:03Z cache-jfk8128 fastly-logs-5-east[285597]: 45.33.64.72 "-" "GET /downloads/point/last-day/ember-phoenix" 429 "npm-registry/0.1.13 node/v4.2.6" "" "MISS" "(null)" "cache-jfk8128-JFK" "339" "213"
<134>2017-05-19T05:21:03Z cache-jfk8128 fastly-logs-5-east[285597]: 45.33.64.72 "-" "GET /downloads/point/last-day/ember-sanctify" 429 "npm-registry/0.1.13 node/v4.2.6" "" "MISS" "(null)" "cache-jfk8128-JFK" "378" "213"
```

The request also results in a burst of `503`s for other API users; because you're saturating backend connections in our CDN.

tldr; I think you might need to use a library like [async](https://www.npmjs.com/package/async) to limit the # of concurrent requests for download counts you make while indexing. I recommend something like:

```js
async.eachLimit(urlsToIndex, 8, () => {})
```

You should still be able to crunch through the download counts fairly fast, and you won't end up eating up all the connections available and getting bad responses back from our API.

Please let me know if there's anything I can do to help, or any more information I can provide.

**Comment by kategengler** [comment]
Oops, very sorry about this 😳 and sorry for the slow reply. I saw the issue but was traveling and unable to respond. We've already begun working on fixing this.

My pedantic side (aka @pgengler) wants to point out that it isn't a **D**DoS since we only have the one server 😝

**Comment by kategengler** [comment]
@bcoe We've deployed the fix for this. Please let us know if everything now looks alright from your end.

**Comment by bcoe** [comment]
@kategengler thanks for getting to this so fast -- nothing to be sorry about; mainly I'm glad I was able to find your website from the IP address in our logs, so that I didn't end up blocking you. The graphs are looking much better:

<img width="985" alt="screen shot 2017-05-21 at 11 20 53 pm" src="https://cloud.githubusercontent.com/assets/194609/26294991/58b2220c-3e7c-11e7-890e-8781119d27ec.png">

For the first time since I started doing a rework of our downloads API, I'm not seeing 503s on a regular interval.

Have an awesome week 👍

**Comment by kategengler** [comment]
Thanks again for letting us know. That code has been doing the same thing since we started 2 years ago, so I guess the rework brought it to light. Have a nice week 😃
